crates/router/src/connector/fiserv/transformers.rs

```diff
@@ -147,6 +187,47 @@ impl TryFrom<&types::ConnectorAuthType> for FiservAuthType {
     }
 }
 
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct FiservCancelRequest {
+    transaction_details: TransactionDetails,
+    merchant_details: MerchantDetails,
+    reference_transaction_details: ReferenceTransactionDetails,
+}
+
+impl TryFrom<&types::PaymentsCancelRouterData> for FiservCancelRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsCancelRouterData) -> Result<Self, Self::Error> {
+        let auth: FiservAuthType = FiservAuthType::try_from(&item.connector_auth_type)?;
+        let metadata = item
+            .connector_meta_data
+            .clone()
+            .ok_or(errors::ConnectorError::RequestEncodingFailed)?;
+        let session: SessionObject = metadata
+            .parse_value("SessionObject")
+            .change_context(errors::ConnectorError::RequestEncodingFailed)?;
+        let cancellation_reason = item
+            .request
+            .cancellation_reason
+            .clone()
+            .ok_or_else(utils::missing_field_err("cancellation_reason"))?;
+        Ok(Self {
+            merchant_details: MerchantDetails {
+                merchant_id: auth.merchant_account,
+                terminal_id: Some(session.terminal_id),
+            },
+            reference_transaction_details: ReferenceTransactionDetails {
+                reference_transaction_id: item.request.connector_transaction_id.to_string(),
+            },
+            transaction_details: TransactionDetails {
+                capture_flag: None,
+                // reversal_reason_code:Some(ReversalReasonCodes::try_from(cancellation_reason)?),
+                reversal_reason_code: Some(cancellation_reason.parse_enum("ReversalReasonCodes").change_context(errors::ConnectorError::MissingRequiredField { field_name: "cancellation reason should be: VOID, TIMEOUT, SUSPECTED_FRAUD, CARD_OVERRIDE" })?)
```

Comment:
Raise `MissingRequiredField` only if data is missing. If the provided data cannot be parsed, raise `InvalidData` error, if it is not present in connector error add it and propogate it.